### DISCUSSION TITLE
Can't make global regular substitution work

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [mikefarah/yq#1747] [Q&A]
My code is ([Nushell](https://www.nushell.sh/)):

```nushell
let code = '    {
        "copyToClipboardCmd": {
            "type": "string"
        },
        "editPreset": {
            "type": "string"
        },
        "edit": {
            "type": "string"
        },
        "editAtLine": {
            "type": "string"
        },
        "editAtLineAndWait": {
            "type": "string"
        },
        "open": {
            "type": "string"
        },
        "openLink": {
            "type": "string"
        }
    }'
$code | yq -P -o json 'to_entries | map(.value.title = (.key | gsub("(?<x>[A-Z])"; " \(.x)") | ascii_downcase)) | from_entries'
```

But it doesn't work: `Error: 1:41: invalid input text "gsub(\"(?<x>[A-Z]..."` is shown. It's not clear what it means: does `gsub` exist at all (but is undocumented), or there is an error inside regex somewhere? If `gsub` doesn't exist I suggest improving the error message to state that there is no such function. My goal is to rewrite this `jq` piece of code:

```
jq 'to_entries | map(.value.title = (.key | gsub("(?<x>[A-Z])"; " \(.x)") | ascii_downcase)) | from_entries'
```

**Comment by mikefarah** [comment]
It's not recognising `gsub` because that hasn't yet been added to `yq` - it's a expression lexing error. It's hard to know from lexing error if its a missing function or other bad syntax.

I'm not 100% sure what your expression does - it seems like it makes keys lower case? In which case you could do:

```
yq 'with_entries(.key |= downcase)' examples/data2.yaml
```

There is a regex `sub` function, documented here: https://mikefarah.gitbook.io/yq/operators/string-operators#sub-regex-replacement - though it doesn't work exactly like `jq` as `yq` is using the build in regex in go...see the docs for more examples.

**Comment by EmilyGraceSeville7cf** [comment]
> I'm not 100% sure what your expression does - it seems like it makes keys lower case?

It was intended to convert all capital letters to lower case and add space before each of them. `blaBla` -> `bla bla`, e.g.
